nilearn/decomposition/multi_pca.py

```diff
@@ -211,7 +230,7 @@ def fit(self, imgs, y=None, confounds=None):
             raise ValueError('Need one or more Niimg-like objects as input, '
                              'an empty list was given.')
         if confounds is None:
-            confounds = itertools.repeat(None, len(imgs))
+            confounds = [None] * len(imgs)  # itertools.repeat(None, len(imgs))
```

Comment:
Could it be because confounds is an iterator that is used twice and in the second case it has already been consumed so you get a StopIteration ? In any case, you should either understand the problem better or remove the commented out code that you changed.